domains/client/domain-executor/src/domain_worker.rs

```diff
@@ -47,6 +47,17 @@ pub(crate) async fn handle_slot_notifications<Block, PBlock, PClient, BundlerFn>
     }
 }
 
+enum BlockImport<BlockImportNotification> {
```

Comment:
this enum seems awkward TBH. Why not just an `Option<BlockInfo>`

Reply:
It can be an Option indeed, the intention was to make it more explicit on the notification source and have some place to leave some comment. I'm okay with Option, which has slightly simpler code, let's see if others feel it helpful.

Reply:
I also agree that being explicit and having some comments on the throttling behavior are important, but it may be better to move the comment to the `maybe_subspace_block_import = block_imports.next()` branch since it is actually `multi_block_import_sender` and `block_import_acknowledgement_sender` performs the throttling, in this way we can also replace `BlockImport<BlockInfo>` with `Option<BlockInfo>` to simpler the code.